### ISSUE TITLE
GCC/Clang error from header files not correctly handled

### ISSUE THREAD

**Issue by blahgeek** [w0rp/ale#985]
Related to #392 

![2017-10-09-16 37 06](https://user-images.githubusercontent.com/1308450/31330321-7709043e-ad10-11e7-88d8-6b0a4f999dfb.png)

The raw output:

```
In file included from <stdin>:3:0:
./common.h:1:13: warning: data definition has no type or storage class
 #define M() syntaxerror
             ^
<stdin>:5:1: note: in expansion of macro ‘M’
./common.h:1:13: warning: type defaults to ‘int’ in declaration of ‘syntaxerror’ [-Wimplicit-int]
 #define M() syntaxerror
             ^
<stdin>:5:1: note: in expansion of macro ‘M’
```

The warning for `./common.h:1:13` should not be marked in the first line of `test.c`

**Comment by w0rp** [comment]
This will probably be fixed as part of the work for #1023

**Comment by w0rp** [comment]
This should be fixed now.
